airflow/models/dag.py

```diff
@@ -44,6 +44,7 @@
     cast,
 )
 
+import cached_property
```

Comment:
We've been using functools.cache_property where available (py 3.8+).  (We should probably put that in a typing_compat or similar helper module so we can not have the try/except in every file that needs it)

Reply:
I'll do that as a separate refactoring PR. This one isn't going to be merged immediately and needs a rebase anyway.